### PR TITLE
chore(ci): update permission

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

Otherwise GitHub cannot create release. Also update planner test to use stream_plan and batch_plan for newly-added queries.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

File: .github/workflow-template/main-override.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ on:
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
+permissions: write-all
```
Comment (fuyufjh) [inline .github/workflow-template/main-override.yml:15]: Looks dangerous... If you want to upload artifacts, take a try on this?

```diff
 permissions:
   id-token: write
   contents: read # This is required for actions/checkout
+  packages: write
``` 

Ref. https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Comment (skyzh) [reply .github/workflow-template/main-override.yml:15]: also need create release permission?

anyway, the current release already contains a bug https://github.com/singularity-data/risingwave/issues/2249. I'll continue to work on this workflow after the holiday. Still hot-patching the release.

Comment (skyzh) [reply .github/workflow-template/main-override.yml:15]: That's great. I'll bump v0.1.7 next week and fix this together.

File: .github/workflow-template/template.yml [hunk 2]
```diff
@@ -16,8 +16,6 @@ env:
   AWS_ROLE_TO_ASSUME: arn:aws:iam::639303875316:role/Create-IAM-Role-for-Configure-AWS-Credentials-Role-1NF1LWROB80QG
 
 permissions:
```
Comment (neverchanje) [inline .github/workflow-template/template.yml:18]:
```suggestion
```

Comment (skyzh) [reply .github/workflow-template/template.yml:18]: I leave it here on purpose, to hint that `permissions` key will be overridden by other yaml files 🤣

**Review by fuyufjh — APPROVED** [review]

**Pull request merged**